src/System.Runtime.Extensions/src/System/AppDomain.cs

```diff
@@ -280,5 +285,37 @@ public void SetThreadPrincipal(IPrincipal principal)
                 _defaultPrincipal = principal;
             }
         }
+
+        public IPrincipal GetThreadPrincipal()
+        {
+            IPrincipal principal = _defaultPrincipal;
+            if (principal == null)
+            {
+                switch (_principalPolicy)
+                {
+                    case PrincipalPolicy.UnauthenticatedPrincipal:
+                        if (s_getUnauthenticatedPrincipal == null)
+                        {
+                            Type type = Type.GetType("System.Security.Principal.GenericPrincipal, System.Security.Claims", throwOnError: true);
+                            Volatile.Write(ref s_getUnauthenticatedPrincipal, (Func<IPrincipal>)Delegate.CreateDelegate(typeof(Func<IPrincipal>), type, "GetDefaultInstance"));
+                        }
+
+                        principal = s_getUnauthenticatedPrincipal();
+                        break;
+
+                    case PrincipalPolicy.WindowsPrincipal:
+                        if (s_getWindowsPrincipal == null)
+                        {
+                            Type type = Type.GetType("System.Security.Principal.WindowsPrincipal, System.Security.Principal.Windows", throwOnError: true);
+                            Volatile.Write(ref s_getWindowsPrincipal, (Func<IPrincipal>)Delegate.CreateDelegate(typeof(Func<IPrincipal>), type, "GetDefaultInstance", ignoreCase: false, throwOnBindFailure: false) ?? throw new PlatformNotSupportedException(SR.PlatformNotSupported_Principal));
```

Comment:
This is ~300 characters long line. Could you please split this to multiple lines?